### DISCUSSION TITLE
Livewire route name parameter is not same like Laravel

### DISCUSSION THREAD

**Discussion by kresnasatya** [livewire/livewire#5161] [1. Help] [answered]
Here's the case e.g name parameter with two words with space.

I create route called `Route::get('/role/{role}', RoleComponent::class)->name('role');`. 

From blade view I put `<a href="{{ route('role', ['role' => rawurlencode('Handsome Admin')]) }}">Role Livewire Component: Handsome admin</a>`.

If I click the link, Livewire will treat like `http://localhost:8000/role/Handsome+Admin`. This is not what I expect.

In Laravel will treat like `http://localhost:8000/role/Handsome%20Admin`. This is what I expect.

How to make Livewire follow the Laravel way for this case?

**Comment by joshhanley** [comment]
@satyakresna yeah there is a difference in the way that Livewire does the URL encoding. But I thought that got fixed. Can you confirm what patch version of Livewire you are using? In your terminal run `composer show -D` and find Livewire.

Thanks!

**Reply by kresnasatya** [reply]
Hi @joshhanley the version Livewire I use is 2.10.6.

**Comment by joshhanley** [comment, marked as answer]
@satyakresna hmm yeah that is the latest. It's possible it didn't get changed as that would now be a breaking change (even though its potentially a bug being different to Laravel).

I'd suggest not using parameters with spaces in them and instead use slugs.

Hope this helps!

**Reply by kresnasatya** [reply]
Thanks @joshhanley for the solution. I was thinking about another way like use the parameter as query string like `?role=Handsome%20Admin`.

**Reply by joshhanley** [reply]
@satyakresna no worries! 🙂 ah yep!
